project/frontend/src/components/ResultsMap.js

```diff
@@ -59,23 +68,37 @@ function MapContainer(props) {
     );
   };
 
+  // TODO: add a spec here to explain what's happening.
   const createMarkers = () => {
-    if (!restaurants) {
+    if (restaurants.length === 0) {
       return;
     }
-    let markers = [];
-    const numOfMarkers = Math.min(restaurants.length, 4);
-    for (let i = 0; i < numOfMarkers; i++) {
-      const coords = restaurants[i].key.latLngCoords;
-      const numInList = (i + 1).toString();
-      markers.push(
-        <MarkerIcon
-          lat={coords.lat}
-          lng={coords.lng}
-          id={i}
-          aria-label={'Your #' + numInList + ' Match'}
-        />
-      );
+    const markers = [];
+    let coords = restaurants[currentCardIndex].key.latLngCoords;
+    markers.push(
+      <MarkerIcon
+        lat={coords.lat}
+        lng={coords.lng}
+        id={currentCardIndex}
+        aria-label={'Your #' + currentCardIndex + ' Match'}
+      />
+    );
+    const numOfMarkers = Math.min(restaurants.length, 3);
+    for (let i = 1; i <= numOfMarkers; i++) {
+      const afterIndex = currentCardIndex + i;
+      if (afterIndex < restaurants.length) {
```

Comment:
for both this check and the before >= 0 check, I think we should abstract away checking if index is valid into a helper function:
```
function isValidIndex(int index):
    return index >= 0 && index < restaurants.length;
```
then you can replace these checks with if isValidIndex(nextCardIndex) & if isValidIndex(prevCardIndex)